ecl/hql/hqlopt.cpp

```diff
@@ -3325,14 +3325,14 @@ IHqlExpression * CTreeOptimizer::doCreateTransformed(IHqlExpression * transforme
             case no_subsort:
                 if (transformedCountProject)
                     break;
-                if (increasesRowSize(transformed))
+                if ((options & HOOminimizeNetworkAndMemory) && increasesRowSize(transformed))
```

Comment:
Moving a project over a sort is often a good idea if that project can be combined with a disk read.

If the rows are definitely held in memory then it doesn't really matter how big the rows that are sorted are - and there is more benefit from combining the project with other operations.  hence the new flag.